Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionReferenceSearchProviders.cs

```diff
@@ -0,0 +1,98 @@
+#if UNITY_EDITOR && UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Search;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    internal static class SearchConstants
+    {
+        // SearchFlags: these flags are used to customize how search is performed and how search
+        // results are displayed in the advanced object picker.
+        // Note: SearchFlags.Packages is not currently used and hides all results from packages.
+        internal static readonly SearchFlags PickerSearchFlags = SearchFlags.Sorted | SearchFlags.OpenPicker;
+
+        // Search.SearchViewFlags : these flags are used to customize the appearance of the PickerWindow.
+        internal static readonly Search.SearchViewFlags PickerViewFlags = Search.SearchViewFlags.OpenInBuilderMode
+            | Search.SearchViewFlags.DisableBuilderModeToggle
+            | Search.SearchViewFlags.DisableInspectorPreview
+            | Search.SearchViewFlags.DisableSavedSearchQuery;
+    }
+
+    internal static class InputActionReferenceSearchProviders
+    {
+        const string k_AssetFolderSearchProviderId = "AssetsInputActionReferenceSearchProvider";
+        const string k_ProjectWideActionsSearchProviderId = "ProjectWideInputActionReferenceSearchProvider";
+
+        // Search provider for InputActionReferences for all assets in the project, without project-wide actions.
+        internal static SearchProvider CreateInputActionReferenceSearchProviderForAssets()
```

Comment:
I'm a little bit torn about overriding the default ADB implementation for this one since it would provide another experience/result than Edit > SearchAll with filter "t:InputActionReference", I guess its fine for pre2 scope but I think we need to ask ourselves if we really want to override actual ADB search for regular assets which are already covered by ADB search. This is the reason I didn't do it on the original branch. Also note that the original branch referenced here didn't use the package search flag if that was a driving reason for doing a custom searcher for assets?

Reply:
Regarding the "incorrect InputAction icon" I think this is the expected behavior for the Search since this is what you get from Advanced Search via main menu. We should follow-up separately what to expect for package assets and sub-asset icons registered via importers. Might be a limitation of the Search functionality which is why I propose it could make sense to not add some custom add on to fix the icons. We should follow-up on this. The reason it fails (after disassembly of the Search code) is that AssetDatabase.CahedIcon doesn't return the icon for the asset if I recall correctly.

I tried to provide a custom IconFetcher on top of ADB before and it didn't work so I think we should reach out to the Search team. Your proposal makes sense.

Reply:
Tracked possible improvements for this here https://jira.unity3d.com/browse/ISX-1681